lib/resque/worker.rb

```diff
@@ -228,7 +228,7 @@ def work(interval = 5.0, &block)
           break if interval.zero?
           log_with_severity :debug, "Sleeping for #{interval} seconds"
           procline paused? ? "Paused" : "Waiting for #{queues.join(',')}"
-          sleep interval
+          @signal_handler.handle_signal(interval)
```

Comment:
As mentioned in the other PR, this will only run if the queue was empty. Should we move this out of the `unless`?